emitter/src/main/java/org/softwire/training/sensoremitter/consumer/SnsTopic.java

```diff
@@ -122,6 +123,8 @@ private void processSubscription(Subscription subscription) {
                 Optional<String> queueUrlO = extractSqsQueueName(subscription.getEndpoint()).flatMap(this::getQueueUrl);
                 if (queueUrlO.isPresent()) {
                     processQueue(subscription, queueUrlO.get());
+                } else if (subscription.getSubscriptionArn().equals(PENDING_CONFIRMATION)) {

```

Comment:
When the stack is created, it creates some [example queues](https://github.com/CorndelWithSoftwire/EventProcessing-Java/blob/Part2/emitter/infrastructure/cloudformation-template.json#L258) (which pick up messages from SNS correctly), and some [other queues called Q1](https://github.com/CorndelWithSoftwire/EventProcessing-Java/blob/Part2/emitter/infrastructure/cloudformation-template.json#L223) which seem to be stuck in the pending state.

This causes the emitter to crash when it first runs the subscription reaper, because the `sns.unsubscribe(subscription.getSubscriptionArn())` line below passes "PendingConfirmation" instead of an ARN.

I think this fix is useful whether or not we get rid of the Q1 queues (or get them into a non-pending state), because we don't want to accidentally delete a student's queue in the brief period where it's pending. Should we also remove them from the CloudFormation template, though?

Reply:
If the example queues aren't useful then I think we should just remove them 👍 

(Unless @SimonStJG can remember if there was some other intended use?)

Reply:
Oops sorry, I think I added the example and the Q1 queues for debugging, I didn't mean to leave them in there.

Reply:
I've deleted the Q1 queues from the CloudFormation template and checked that the stack can still be created. 👍